extract_pov_edits/dataflow_main.py

```diff
@@ -178,48 +209,97 @@ def process(self, element, bucket, ingestFrom):
          raise Exception("GSUTIL COPY Error, exited with status %d" % status)
        input_stream = chunk_name
     # Running ingestion on the xml file
-    last_revision = 'None'
+    last_revision = None
     last_completed = time.time()
-    cur_sents = {}
     i = 0
     for i, content in enumerate(parse_stream(bz2.BZ2File(chunk_name))):
-      # List of X on Wikipedia includes long tables, might be the reason that
-      # mwparser crushes
-      if 'list' in content['page_title'].lower():
+      if (page_ids is not None) and (content['page_id'] not in page_ids):
         continue
       self.processed_revisions.inc()
-      last_revision = content['rev_id']
-      # Add the week and year field for sharding
       if content['text'] is None:
         content['text'] = ""
-      yield content
-      (context_equals, inserts, deletes, cur_sents), error = process(content,cur_sents)
-      if error:
-        yield beam.pvalue.TaggedOutput('error_log', json.dumps(content['rev_id']))
-        self.errors.inc()
-      metadata = {f : content[f] for f in ['comment', 'user_id', 'user_text',
-                                           'user_ip', 'page_id', 'page_title']}
-      if content["comment"] is not None and "POV" in content["comment"]:
-        rejections = {"rejecter" : content['rev_id'],
-                      "rejectee" : [d[1] for d in deletes]}
-        rejections.update(metadata)
-        yield beam.pvalue.TaggedOutput('pov_rejections', json.dumps(rejections))
-        ret = {"rejecter" : content['rev_id']}
-        ret.update(metadata)
-        for d in deletes:
-          ret['content'] = d[0]
-          ret['rejectee'] = d[1]
-          yield beam.pvalue.TaggedOutput('pov_rejects', json.dumps(ret))
-        for sent in inserts:
-          ret['content'] = sent[0]
-          yield beam.pvalue.TaggedOutput('npov_improvements', json.dumps(ret))
-        for sent in context_equals:
-          ret['content'] = sent
-          yield beam.pvalue.TaggedOutput('pov_non_rejects', json.dumps(ret))
-      logging.info('CHUNK {chunk}: revision {revid} ingested, time elapsed: {time}.'.format(chunk=chunk_name, revid=last_revision, time=time.time() - last_completed))
+      if (content["comment"] is not None and "POV" in content["comment"]) or all_edits:
+        yield json.dumps((last_revision, content))
+      logging.info('INGESTION_LOG: CHUNK {chunk}: revision {revid} ingested, time elapsed: {time}.'.format(
```

Comment:
don't use .format() in LOG statements.
logger supports passing dictionary of via extra
https://docs.python.org/2/library/logging.html#logging.Logger.debug